pkg/cmd/image/list.go

```diff
@@ -188,50 +189,126 @@ type imagePrinter struct {
 	quiet, noTrunc, digestsFlag, namesFlag bool
 	tmpl                                   *template.Template
 	client                                 *containerd.Client
-	contentStore                           content.Store
+	contentStore                           containerdutil.Provider
 	snapshotter                            snapshots.Snapshotter
 }
 
-func (x *imagePrinter) printImage(ctx context.Context, img images.Image) error {
-	ociPlatforms, err := images.Platforms(ctx, x.contentStore, img.Target)
+type Image struct {
+	blobSize int64
+	size     int64
+	platform platforms.Platform
+	config   *ocispec.Descriptor
+}
+
+var ReadBlob = containerdutil.WithCache()
+
+func ReadManifest(ctx context.Context, provider containerdutil.Provider, snapshotter snapshots.Snapshotter, desc ocispec.Descriptor) (*Image, error) {
+	// Read the manifest blob from the descriptor
+	manifestData, err := ReadBlob(ctx, provider, desc)
 	if err != nil {
-		log.G(ctx).WithError(err).Warnf("failed to get the platform list of image %q", img.Name)
-		return x.printImageSinglePlatform(ctx, img, platforms.DefaultSpec())
+		return nil, err
 	}
-	psm := map[string]struct{}{}
-	for _, ociPlatform := range ociPlatforms {
-		platformKey := makePlatformKey(ociPlatform)
-		if _, done := psm[platformKey]; done {
-			continue
-		}
-		psm[platformKey] = struct{}{}
-		if err := x.printImageSinglePlatform(ctx, img, ociPlatform); err != nil {
-			log.G(ctx).WithError(err).Warnf("failed to get platform %q of image %q", platforms.Format(ociPlatform), img.Name)
-		}
+
+	// Unmarshal as Manifest
+	var manifest ocispec.Manifest
+	if err := json.Unmarshal(manifestData, &manifest); err != nil {
+		return nil, err
 	}
-	return nil
+
+	// Now, read the config
+	configData, err := ReadBlob(ctx, provider, manifest.Config)
+	if err != nil {
+		return nil, err
+	}
+
+	// Unmarshal as Image
+	var config ocispec.Image
+	if err := json.Unmarshal(configData, &config); err != nil {
+		log.G(ctx).Error("Error unmarshaling config")
+		return nil, err
+	}
+
+	// If we are here, the image exists and is valid, so, do our size lookups
+
+	// Aggregate the blob size from the config and layers
+	blobSize := manifest.Config.Size
```

Comment:
Can we add the size of manifest blob as well?

Reply:
Yes, I missed this one.
Thanks for catching it.
Next push will have it.